ballerina/tests/parse_yaml_streams.bal

```diff
@@ -0,0 +1,289 @@
+// under the License.
+
+import ballerina/io;
+import ballerina/test;
+
+const YAML_STREAM_TEST_PATH = FILE_PATH + "streams/";
+
+@test:Config
+isolated function testYamlStringParsing() returns error? {
+    stream<io:Block, io:Error?> streamResult = check io:fileReadBlocksAsStream(YAML_STREAM_TEST_PATH + "stream_1.yaml");
+    anydata result = check parseStream(streamResult);
+    test:assertTrue(result is anydata[]);
+    test:assertEquals((<anydata[]>result).length(), 4);
+}
+
+@test:Config
+isolated function testYamlStringParsing2() returns error? {
+    string filePaht = YAML_STREAM_TEST_PATH + "stream_1.yaml";
+    stream<io:Block, io:Error?> streamResult = check io:fileReadBlocksAsStream(filePaht);
+    ExpectedType result = check parseStream(streamResult);
+
+    final ConfigType configMapValue = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {"name": "ballerina-mongodb-configmap"},
+        "data": {
+            "Config.toml": "[ballerina.ballerina_mongodb_kubernetes]" +
+                            "\nmongodbPort = 27017\nmongodbHost = \"mongodb-service\"" +
+                            "\ndbName = \"students\"\n\n[ballerina.log]\nlevel = \"DEBUG\"\n"
+        }
+    };
+
+    test:assertEquals(result.length(), 2);
+    test:assertEquals(result[1].apiVersion, configMapValue.apiVersion);
+    test:assertEquals(result[1].kind, configMapValue.kind);
+    test:assertEquals(result[1].metadata, configMapValue.metadata);
+    test:assertEquals((<ConfigType>result[1]).data, configMapValue.data);
+}
+
+@test:Config
+isolated function testYamlStringParsing3() returns error? {
+    string filePaht = YAML_STREAM_TEST_PATH + "stream_1.yaml";
+    stream<io:Block, io:Error?> streamResult = check io:fileReadBlocksAsStream(filePaht);
+    UnionType[] result = check parseStream(streamResult);
+
+    final ConfigType configMapValue = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {"name": "ballerina-mongodb-configmap"},
+        "data": {
+            "Config.toml": "[ballerina.ballerina_mongodb_kubernetes]" +
+                            "\nmongodbPort = 27017\nmongodbHost = \"mongodb-service\"" +
+                            "\ndbName = \"students\"\n\n[ballerina.log]\nlevel = \"DEBUG\"\n"
+        }
+    };
+
+    test:assertEquals(result.length(), 4);
+    test:assertEquals(result[1].apiVersion, configMapValue.apiVersion);
+    test:assertEquals(result[1].kind, configMapValue.kind);
+    test:assertEquals(result[1].metadata, configMapValue.metadata);
+    test:assertEquals((<ConfigType>result[1]).data, configMapValue.data);
+}
+
+@test:Config
+isolated function testYamlStreamPastingWithTupleExpected() returns error? {
+    string filePaht = YAML_STREAM_TEST_PATH + "stream_1.yaml";
+    stream<io:Block, io:Error?> streamResult = check io:fileReadBlocksAsStream(filePaht);
+    [ServiceType, ConfigType, ConfigType, DeploymentType] result = check parseStream(streamResult);
```

Comment:
Do you have test with `strictTupleOrder` option set to true?

Reply:
added in [commit](https://github.com/ballerina-platform/module-ballerina-data.yaml/pull/4/commits/2b1ea84485f3dc3e76c71fffa9027226b7faf72f).